shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -370,6 +370,16 @@ private void setSystemChromeSystemUIOverlayStyle(
     WindowInsetsControllerCompat windowInsetsControllerCompat =
         new WindowInsetsControllerCompat(window, view);
 
+    // Flag set to specify that this window is responsible for drawing the background for the system
+    // bars.
+    window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+
+    // Flag set to dismiss any requests for translucent system bars to be provided in lieu of what
+    // is specified by systemChromeStyle.
+    window.clearFlags(
+        WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS
+            | WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION);
+
```

Comment:
These flags are deprecated in API 30

Reply:
They are needed to fix the problem for API < 30, so I added a comment noting the deprecation and suppressed the warning